pkg/workspace/workspace.go

```
@@ -134,20 +135,30 @@ func (ws *Workspace) CatalogBasePath() string {
 	)
 }
 
-// Returns the catalog path for catalog with a given name within a workspace
-func (ws *Workspace) CatalogPath(name *string) string {
-	if name == nil {
-		return filepath.Join(
-			ws.rootPath,
-			".warpforge",
-			"catalog",
-		)
-	} else {
-		return filepath.Join(
-			ws.CatalogBasePath(),
-			*name,
-		)
+// defaultCatalogPath returns the path to the default catalog belonging to this workspace.
+func (ws *Workspace) defaultCatalogPath() string {
+	return filepath.Join(ws.rootPath, ".warpforge", "catalog")
+}
+
+// Returns the catalog path for catalog with a given name within a workspace.
+// Guards against filepath modifying names by considering any path which
+// would be modified by filepath.Clean to be considered invalid.
+// Guards against catalog nesting by considering any name with filepath separators to be invalid.
+//
+// Errors:
+//
+//    - warpforge-error-catalog-invalid -- when catalog name is invalid
+func (ws *Workspace) CatalogPath(name string) (string, wfapi.Error) {
+	if name == "" {
+		return ws.defaultCatalogPath(), nil
 	}
```

This should return an error if we're in a root workspace, come to think of it.